doc/source/development/rfc/rfc99_geometry_coordinate_precision.rst

```diff
@@ -0,0 +1,674 @@
+.. _rfc-99:
+
+===================================================================
+RFC 99: Geometry coordinate precision
+===================================================================
+
+============== =============================================
+Author:        Even Rouault
+Contact:       even.rouault @ spatialys.com
+Started:       2024-Feb-24
+Status:        In development
+Target:        GDAL 3.9
+============== =============================================
+
+Summary
+-------
+
+This RFC aims at introducing optional metadata to specify the coordinate
+precision of geometries, to be able to round appropriately coordinates and limit
+the number of decimals when exporting to text-based formats or nullify
+least-significant bits for binary formats. That metadata will be stored into
+and read from formats that can support it.
+
+Motivation
+----------
+
+The aim is multiple:
+
+- reducing file size. For text-based formats, rounding and truncating to the
+  specified precision directly reduce file size. For binary formats, using that
+  information to zero least-significant bits can increase the potential when
+  applying afterwards lossless compression methods (typically zipping a file).
+
+- presenting the user with hints on the precision of the data he/she accesses.
+  This can be used by user interfaces build on top of GDAL to display geometry
+  coordinates with an appropriate number of decimals.
+
+- a few drivers (GeoJSON, JSONFG, OpenFileGDB) have layer creation options to
+  specify coordinate precision, but there is currently no driver agnostic way
+  of specifying it.
+
```

Comment:
Not sure if this is relevant, but just for information. 

The motivation why I'm interested in applying precisions in general... and this feature can play a role in standardising/specifying the precision at the layer/dataset level in a clear way, is to avoid slivers. When calculating several overlays between different layers after each other, or if calculating overlays between layers where features have been snapped on line segments of the other layer, slivers or dangling nodes of ~1e-8 (m) or narrower will appear in the result due to rounding issues. Using a GEOS set_precision of >=1e-8 will remove such dangling nodes from larger polygons and will change sliver polygons to EMPTY geometries or NULL, making it easy to identify and remove them.